pyteal/ast/assert_.py

```diff
@@ -12,27 +13,39 @@
 class Assert(Expr):
     """A control flow expression to verify that a condition is true."""
 
-    def __init__(self, cond: Expr, *additional_conds: Expr) -> None:
+    def __init__(
+        self, cond: Expr, *additional_conds: Expr, comment: str = None
+    ) -> None:
         """Create an assert statement that raises an error if the condition is false.
 
         Args:
             cond: The condition to check. Must evaluate to a uint64.
             *additional_conds: Additional conditions to check. Must evaluate to uint64.
+            comment: String comment to place on the line immediately prior to the assert op
         """
         super().__init__()
         require_type(cond, TealType.uint64)
         for cond_single in additional_conds:
             require_type(cond_single, TealType.uint64)
+
+        self.comment = comment
         self.cond = [cond] + list(additional_conds)
 
     def __teal__(self, options: "CompileOptions"):
         if len(self.cond) > 1:
-            asserts = [Assert(cond) for cond in self.cond]
+            asserts: list[Expr] = []
+            for cond in self.cond:
+                asrt = Assert(cond, comment=self.comment)
+                asrt.trace = cond.trace
```

Comment:
trace is useful for the error messages, when we create a new Expr within pyteal the trace gets set to the internal pyteal file rather than the authors source file (which is probably what they actually want)

By setting the trace here to the conds trace, we're passing the info about where it was initially declared to be at least much closer to the location it was declared in the authors source

Reply:
I see, that makes sense.